---
layout: default
---

### **Welcome to CVR 6th Annual training course on Viral Bioinformatics and Genomics course (21 - 25 August 2023)**
##### **Location: [Boyd Orr building](https://goo.gl/maps/MxN5UhB9ukpgx3sB7), University of Glasgow, United Kingdom**

### Course content

The 2023 course will introduce the participants to the UNIX OS and Bash scripting along with a suite of bioinformatics tools covering the following topics: 

▪ HTS sequencing technologies: overview of the different HTS platforms and sample preparations \
▪ Unix OS: introduction to Unix OS and essential Bash scripting \
▪ Reference assembly: aligning sequence reads to a known reference and visualization (e.g., bowtie2, BWA, Tanoti, Tablet, UGENE) \
▪ Variant calling: consensus sequence generation, low frequency variant calling and error correction (e.g., samtools, LoFreq, DiversiTools) \
▪ De-novo assembly: overlap layout and de Bruijn graphs approaches for sequence assembly, quality assessment and merging contigs (e.g., ABySS, SPAdes, MIRA, IDBA-UD, QUAST) \
▪ Metagenomic analyses:sanitizing sequence datasets, assembling, annotating, visualization (e.g., MetAMOS, Krona, DIAMOND, Kraken) \
▪ Genomics: scaffolding, improving and finishing the assembly, gene annotation (e.g., ICORN, Artemis, RATT) \
▪ Phylogenetic analysis: introduction to multiple sequence alignment and phylogenetic reconstruction (e.g., Muscle, PhyML, FigTree) 

### Prerequisites
Preference will be given to applicants who: (1) have some familiarity with HTS technologies; (2) have already or are planning to generate viral HTS data in their work; and (3) have an interest in computers and programming (some basic experience in a command-line environment is a prerequisite)

### Instructors

[David Robertson](https://www.gla.ac.uk/schools/infectionimmunity/staff/davidrobertson/) \
[Derek Wright](https://www.gla.ac.uk/schools/infectionimmunity/staff/derekwright/) \
[Joseph Hughes](https://www.gla.ac.uk/schools/infectionimmunity/staff/josephhughes/) (Course Organizer) \
[Quan Gu](https://www.gla.ac.uk/schools/infectionimmunity/staff/quangu/) \
[Richard Orton](https://www.gla.ac.uk/schools/infectionimmunity/staff/richardorton/) \
[Sreenu Vattipally](https://www.gla.ac.uk/schools/infectionimmunity/staff/sreenuvattipally/) \
[Srikeerthana Kuchi](https://www.gla.ac.uk/schools/infectionimmunity/staff/srikeerthanakuchi/) (Course Organizer)

### [Timetable](https://drive.google.com/file/d/18wkXigU7Jx8VMeZsBAZ6OtmRVr1tn_NM/view?usp=share_link)

### Practical Modules
1. [Introduction to Linux](docs/Introduction_to_Linux.md)
2. [Read cleaning](docs/ReadCleaning.md)
3. [Reference Alignment](docs/ReferenceAlignment.md)
4. [Phylogenetics](docs/Phylogenetics.md)
5. [De novo assembly](docs/DeNovoPracticals.md)
6. [De novo quality assessment](docs/QualityAssessment.md)
7. [Metagenomics and virus discovery](docs/Metagenomics.md)
8. [K-mer based metagenomic profiling](docs/KmerMetagenomics.md)
9. [Introduction to Transcriptomics](docs/Transcriptomics.md)





